src/core/jpx.js

```diff
@@ -24,20 +24,101 @@ class JpxError extends BaseException {
 }
 
 class JpxImage {
-  static #module = null;
-
-  static decode(data, decoderOptions) {
-    decoderOptions ||= {};
-    this.#module ||= OpenJPEG({ warn });
-    const imageData = this.#module.decode(data, decoderOptions);
-    if (typeof imageData === "string") {
-      throw new JpxError(imageData);
+  static #buffer = null;
+
+  static #handler = null;
+
+  static #instantiationFailed = false;
+
+  static #modulePromise = null;
+
+  static #openjpegWasmUrl = "";
+
+  static setOptions({ handler, openjpegWasmUrl }) {
+    if (!this.#buffer) {
+      this.#openjpegWasmUrl = openjpegWasmUrl;
+      if (!openjpegWasmUrl) {
+        this.#handler = handler;
+      }
+    }
```

Comment:
Why do we need the `this.#buffer` check here, since that's not immediately obvious to me?

Reply:
Once we've a buffer we don't need get it again so those options are useless.